libsolidity/ast/Types.cpp

```diff
@@ -1971,6 +1971,19 @@ string ArraySliceType::toString(bool _short) const
 	return m_arrayType.toString(_short) + " slice";
 }
 
+TypeResult ArraySliceType::interfaceType(bool) const
```

Comment:
Doesn't interfaceType only make sense for types that can be "written down" because it is used for types of function parameters (not arguments)?

How is it done for string literals? mobileType? encodingType?

Reply:
Alternative would be ``mobileType()`` yes, I can do that.
I actually wanted to use ``encodingType()`` only, but the logic is:
```
	TypePointer encodingType = mobileType();
	if (encodingType)
		encodingType = encodingType->interfaceType(_inLibraryCall);
	if (encodingType)
		encodingType = encodingType->encodingType();
```
So that doesn't work. But I'll change to ``mobileType()``.

Reply:
Did that now.